modules/api_gateway/main.tf

```diff
@@ -0,0 +1,151 @@
+
+data "aws_region" "current" {}
+data "aws_caller_identity" "current" {}
+
+data "aws_lambda_function" "async" {
+  function_name = var.async_lambda_name
+}
+
+# create the role that API Gateway can use to call Step Functions.
+data "aws_iam_policy_document" "trust" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["apigateway.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "api_gw" {
+  statement {
+    actions   = ["lambda:InvokeFunction"]
+    resources = [data.aws_lambda_function.async.arn]
+  }
+}
+
+resource "aws_iam_policy" "api_gw" {
+  name        = "${var.project_name}-api-gw"
+  description = "Policy used by the Ldap Maintenance API Gateway"
+  policy      = "${data.aws_iam_policy_document.api_gw.json}"
+}
+
+resource "aws_iam_role" "api_gw" {
+  name = "${var.project_name}-api-gw"
+
+  assume_role_policy = data.aws_iam_policy_document.trust.json
+
+  tags = var.tags
+}
+
+resource "aws_iam_policy_attachment" "api_gw" {
+  name       = "ldap-maintainer-api-gw"
+  roles      = ["${aws_iam_role.api_gw.name}"]
+  policy_arn = "${aws_iam_policy.api_gw.arn}"
+}
+
+resource "aws_api_gateway_rest_api" "api" {
+  name        = "${var.project_name}-api"
+  description = "API for managing LDAP maintenance tasks"
+}
+
+resource "aws_api_gateway_resource" "event_listener" {
+  rest_api_id = "${aws_api_gateway_rest_api.api.id}"
+  parent_id   = "${aws_api_gateway_rest_api.api.root_resource_id}"
+  path_part   = "event-listener"
+}
+
+resource "aws_api_gateway_method" "event_listener_post" {
+  rest_api_id   = "${aws_api_gateway_rest_api.api.id}"
+  resource_id   = "${aws_api_gateway_resource.event_listener.id}"
```

Comment:
rest_api_id and resource_id doesn't need the $ or even the `""`

Reply:
updated